### ISSUE TITLE
Export distilled models for aprender inference consumption

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#12]
## Summary

Coordinate entrenar's distillation output with aprender's inference ingestion.

## Context

aprender needs to consume distilled/fine-tuned models from entrenar for:
- depyler oracle (error prediction)
- ruchy/bashrs transpiler oracles
- Cross-transpiler transfer learning

See: `aprender/docs/specifications/more-learning-specs.md` §34

### 1. LoRA/Adapter Export

```rust
// aprender expects this in .apr format
pub struct AdapterConfig {
    pub rank: usize,                    // 4, 8, 16
    pub alpha: f32,
    pub target_modules: Vec<String>,    // ["q_proj", "v_proj"]
}

pub struct AdapterWeights {
    pub lora_a: HashMap<String, Tensor>,  // down-projection
    pub lora_b: HashMap<String, Tensor>,  // up-projection
}
```

**Request:** Export adapters as separate `.apr` files with `ModelType::Adapter`

### 2. Training Metadata

```rust
pub struct TrainingMetadata {
    pub trained_by: String,           // "entrenar-v0.5.0"
    pub training_steps: u64,
    pub final_metrics: HashMap<String, f32>,
    pub hardware: String,             // "A100-80GB"
    pub training_time_secs: u64,
}
```

**Request:** Include in .apr metadata section

### 3. Distillation Provenance

Already in .apr spec, but please ensure:
- `teacher.hash` is SHA256 of teacher weights
- `method` is one of: Standard, Progressive, Ensemble
- `params.final_loss` is populated
- `params.temperature` used during distillation

### 4. Checkpoint Format (Future)

For resumable distillation:
- Optimizer state
- Learning rate scheduler state
- Current epoch/step
- RNG state for reproducibility

## Integration Points

| entrenar produces | aprender consumes |
|-------------------|-------------------|
| `student.apr` | `load()` for inference |
| `adapter.apr` | `load_adapter()` + merge |
| `checkpoint.apr` | Resume training (future) |

## Acceptance Criteria

- [ ] LoRA adapters export to `.apr` with AdapterConfig
- [ ] TrainingMetadata populated in distilled models
- [ ] Teacher hash verified against known models
- [ ] Integration test: entrenar distill → aprender infer

## Related

- aprender#101 - MoE implementation (done)
- entrenar#11 - MoE format support
- `more-learning-specs.md` §33 (Transfer Learning), §34 (Distillation Ingestion)

**Comment by noahgift** [comment]
Implemented in src/hf_pipeline/export/ (GGUF writer, export pipeline, weight conversion, publishing) and src/hf_pipeline/distillation/ (attention distillation, loss, progressive distillation). Also src/distill/ provides core KD loss and checkpoint support for inference consumption.
